src/NServiceBus.Core.Tests/Pipeline/MutateInstanceMessage/MutateIncomingMessageBehaviorTests.cs

```diff
@@ -13,10 +13,12 @@ class MutateIncomingMessageBehaviorTests
         [Test]
         public async Task Should_not_call_MutateIncoming_when_hasIncomingMessageMutators_is_false()
         {
-            var behavior = new MutateIncomingMessageBehavior(hasIncomingMessageMutators: false);
+            var behavior = new MutateIncomingMessageBehavior();
 
             var context = new TestableIncomingLogicalMessageContext();
 
+            await behavior.Invoke(context, ctx => TaskEx.CompletedTask);
```

Comment:
This test now calls the invoke twice and moves the registration before the second invoke. This is not exactly how it would be in real life since containers are usually closed for modification after the first resolve. I figured it is good enough for this test and the following as well.